.github/workflows/on-commit.yml

```diff
@@ -31,51 +39,40 @@ jobs:
             set -xe
             mount --bind /cvasl/test_data/ /ci/test_data/
             cd /ci
-            python -m pip install pycodestyle "pytest<8.1" nbmake
-            python ./setup.py bdist_wheel
-            python -m pip install ./dist/*.whl
-            python ./setup.py test'
-
+            python -m pip install tox
+            python -m tox -e  ${{ matrix.tox-python }} \
+            -- tests/test_core.py'
   SanityOnMulti:
-    runs-on: ${{ matrix.os }}
-    strategy:
-      matrix:
-        os: [ubuntu-latest, macos-latest]
-        python-version: ['3.10', '3.11']
-    steps:
-      - uses: actions/checkout@v3
-        with:
-          submodules: recursive
-      - uses: s-weigand/setup-conda@v1
-        with:
-          conda-channels: conda-forge
-          python-version: '3.11'
-      - run: python3 -m venv .venv
-      - run: .venv/bin/python -m pip install pycodestyle "pytest<8.1" nbmake
-      - run: .venv/bin/python ./setup.py bdist_wheel
-      - run: .venv/bin/python -m pip install ./dist/*.whl
-      - run: .venv/bin/python ./setup.py test -a "test_multi.py"
-      # needs to be updated to include Windows
+      runs-on: ${{ matrix.os }}
+      strategy:
+        matrix:
+          os: [ubuntu-latest, macos-latest, windows-latest]
+          python-version: ['3.10', '3.11']
+          include:
+            - python-version: '3.10'
+              tox-python: "py310"
+            - python-version: '3.11'
+              tox-python: 'py311'
+            - os: windows-latest
+              python: .venv/Scripts/python.exe
+            - os: ubuntu-latest
+              python: .venv/bin/python
+            - os: macos-latest
+              python: .venv/bin/python
```

Comment:
I was a bit confused by this at first. I see what it's trying to do, but using venvs this way is a bit risky and may break, because it's not intended to be used like this. The default thing to do is to "activate" the venv, see the commands in the table here: https://docs.python.org/3/library/venv.html#how-venvs-work

Now, in this case you still need two different commands for Windows on the one hand and Linux/macOS on the other. To cut down on complexity of this workflow, I would suggest to not code these commands here, but to do it below by making the steps dependent on the platform. See the suggestion in a next review comment below for line 70.

This section can then be removed.

```suggestion
```

Reply:
Every run step potentially starts a new shell session. COde source in previous  run step may  not transfer to the next. Therefore trying to activate environment as done interactively is not expected to work.

Reply:
Ah right, super annoying.

An alternative btw is to not use a venv at all. Most Python packages I know just setup Python using the setup-python Action, like for instance in the Python template here: https://github.com/NLeSC/python-template/blob/455ae35bd67e7ddb63cdeff184da4759a81e5a5d/%7B%7Bcookiecutter.directory_name%7D%7D/.github/workflows/build.yml#L23

Unless they use Conda, but in your case you seem to use Conda only to install a Python version, which could also be done by the conda-less setup-python Action. Either way, I think it should also with Conda be possible to go without a venv...

Anyway, something to try out next PR perhaps.